src/patternfly/components/Menu/menu.scss

```diff
@@ -489,298 +385,217 @@
   }
 
+  // - Menu item loading
   &.pf-m-loading {
-    --#{$menu}__list-item--hover--BackgroundColor: transparent;
-    --#{$menu}__list-item--focus-within--BackgroundColor: transparent;
-
     justify-content: center;
-    padding-block-start: var(--#{$menu}__list-item--m-loading--PaddingTop);
     overflow: hidden; // prevents spinner rotation from overflowing
+    pointer-events: none;
+
+    .#{$menu}__item-text {
+      text-align: center;
+    }
   }
 
+  // - Menu item danger
   &.pf-m-danger {
-    --#{$menu}__item--Color: var(--#{$menu}__list-item--m-danger__item--Color, inherit);
+    --#{$menu}__item--Color: var(--#{$menu}__list-item--m-danger--Color);
+
+    &:hover,
+    &:focus {
+      --#{$menu}__item--Color: var(--#{$menu}__list-item--m-danger--hover--Color, var(--#{$menu}__list-item--m-danger--Color));
+    }
+  }
+
+  &:has(.#{$menu}__item-action) {
+    padding-inline-end: var(--#{$menu}__list-item--has--menu-action--PaddingInlineEnd);
   }
 }
 
-// Item
+// TODO: organize these in order of importance
+// - Menu item
 .#{$menu}__item {
-  display: flex;
   flex-basis: 100%;
   flex-direction: column;
   min-width: 0;
-  padding-block-start: var(--#{$menu}__item--PaddingTop);
-  padding-block-end: var(--#{$menu}__item--PaddingBottom);
-  padding-inline-start: var(--#{$menu}__item--PaddingLeft);
-  padding-inline-end: var(--#{$menu}__item--PaddingRight);
+  padding-block-start: var(--#{$menu}__item--PaddingBlockStart, var(--#{$menu}__item--PaddingBlock));
+  padding-block-end: var(--#{$menu}__item--PaddingBlock, var(--#{$menu}__item--PaddingBlock));
+  padding-inline-start: var(--#{$menu}__item--PaddingInlineStart, var(--#{$menu}__item--PaddingInline));
+  padding-inline-end: var(--#{$menu}__item--PaddingInlineEnd, var(--#{$menu}__item--PaddingInline));
   font-size: var(--#{$menu}__item--FontSize);
   font-weight: var(--#{$menu}__item--FontWeight);
   line-height: var(--#{$menu}__item--LineHeight);
   color: var(--#{$menu}__item--Color);
   text-align: start;
+  text-decoration: none;
   background-color: var(--#{$menu}__item--BackgroundColor);
-  border: none;
-
-  &:hover {
-    text-decoration: none;
-  }
+  border: 0;
+  outline-offset: var(--#{$menu}--OutlineOffset);
 
-  &:hover,
-  &:focus {
-    --#{$menu}__item-external-icon--Opacity: 1;
+  &:is(:hover, :focus) {
+    --#{$menu}__item--BackgroundColor: var(--#{$menu}__list-item--hover--BackgroundColor);
   }
 
-  &.pf-m-selected {
+  &:is(.pf-m-selected, :hover) {
     .#{$menu}__item-select-icon {
       opacity: 1;
     }
   }
 
-  @at-root label#{&}:where(:not([disabled], .pf-m-disabled, .pf-m-aria-disabled)) {
-    cursor: pointer;
+  &.pf-m-selected {
+    --#{$menu}__item-select-icon--Color: var(--#{$menu}__item-select-icon--m-selected--Color);
   }
+
 }
 
+// - Menu item select icon
+.#{$menu}__item-select-icon {
+  color: var(--#{$menu}__item-select-icon--Color, var(--#{$menu}--link--icon--Color));
```

Comment:
I don't know if using `--#{$menu}--link--icon--Color` is best here, since the icon is just blue, and not a link. Also I see `--#{$menu}--link--icon--Color` is used on the external link icon, and I don't know that those share a concept for a shared var, they seem to just both be blue? WDYT?